### ISSUE TITLE
Running Goofys in the foreground is the only way to prevent transport endpoint disconnections from occuring

### ISSUE THREAD

**Issue by BnJam** [kahing/goofys#634]
I am mounting a Google Cloud Storage bucket with goofys (fuse) to my docker container and running deep learning training. 

The training data is ~10k datapoints and test ~600 datapoints. Between model fitting and testing, I rsync the model into the bucket since the bucket doesn't support directly writing HDF5 format, but that's something else. 
When running goofys in the background, the application crashes when attempting to access the test data. So it handles all the training data just fine and copies the model over to the bucket. When the testing begins, I am met with the output: 

```bash
Transport endpoint is not connected
```

I am mounting with the following command (for background mounting, foreground has the `-f` flag):

```bash
goofys -o allow_other --profile <profilename> --stat-cache-ttl 10s --type-cache-ttl 10s --endpoint <endpnt> --dir-mode 0777 --file-mode 0777 <source_bucket> /mnt/<src_dir>
```
If I mount the bucket in the foreground, the application runs through in its entirety. 

I don't understand why it would work in the foreground, but not in the background. What difference is there between the daemon versus the goofys app when there is a very high throughput of data? 

goofys version: goofys version 0.24.0-45b8d78375af1b24604439d2e60c567654bcdf88

**Comment by BnJam** [comment]
I "fixed" this issue by using GSCFuse instead of Goofys, unfortunately. 

Would like to understand why running Goofys in the background (but not the foreground) causes Fuse transport connection terminating.

**Comment by kahing** [comment]
there isn't much info here. What does your dockerfile look like? how did you start goofys exactly? do you have debug logs?

**Comment by hameer-spire** [comment]
I'm having the exact same issue. I can't get goofys to work or mount stuff in the background, running a simple `ls` in a goofys mounted directory hangs.

Unfortunately, the test is in docker, and I can't find a way to get syslogs within docker.

I'm using the following invocation:

```bash
function mount_goofys {
    if [[ ! -d "$2" ]]; then
        mkdir -p "$2"
        goofys -o allow_other --cache "--free:30%:/tmp" --debug_fuse --debug_s3 "$1" "$2"
    fi
}
```

**Comment by arttkach** [comment]
Similar issue here. I'm running goofys in a docker container to mount an s3 bucket. Once mounted, I'm not able to write to files. Running with `-f` flag works. No syslog due to running inside a container. Tried @hameer-spire suggestion with` -v /dev/log:/dev/log` but that didn't work for me.
With `-f` flag I see some errors get printed but still able to write to file in my s3.
`(base) jovyan@ed931d10617b:~$ goofys -f data-fabric-test1 data-fabric-test1
2023/01/23 19:11:13.210802 s3.INFO Switching from region 'us-east-1' to 'us-west-2'
2023/01/23 19:11:13.548120 main.INFO File system has been successfully mounted.
2023/01/23 19:11:32.095919 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:11:32.096876 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:11:33.382382 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:11:33.383271 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:12:54.891687 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:12:54.892649 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:13:16.578336 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:13:16.578875 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:13:17.091615 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
2023/01/23 19:13:17.092669 fuse.ERROR *fuseops.SetXattrOp error: operation not permitted
`
I can post my Dockerfile if needed.

**Comment by hameer-spire** [comment]
I got this to work by installing `rsyslog` inside the container and making it priveleged.

**Comment by arttkach** [comment]
Yep, I run my container in privileged mode as well. Otherwise, /dev/fuse is not getting mounted.
I tried installing rsyslog but still not able to write to files in s3 without `-f` flag.
After a write attempt error I do ls and I get:
`ls: cannot open directory '.': Transport endpoint is not connected`.
I added rsyslog to other dependencies in my Dockerfile. My container runs Ubuntu 22.04.1
@hameer-spire how did you install it?
